client/dashboard/sites/site-reset-modal/index.tsx

```diff
@@ -0,0 +1,252 @@
+				} else {
+					onError( __( 'The site could not be reset due to an error. Please try again.' ) );
+				}
+			},
+		} );
+	};
+
+	return (
+		<VStack spacing={ 6 }>
+			<Notice variant="warning">
+				<Text>
+					{ createInterpolateElement(
+						__(
+							'Before resetting your site, consider <link>exporting your content as a backup</link> ↗.'
+						),
+						{
+							link: <a href={ `/export/${ siteDomain }` } />,
+						}
+					) }
+				</Text>
+			</Notice>
+			<Text as="p">
+				{ sprintf(
+					/* translators: %s: site domain */
+					__(
+						"Resetting %s will remove all of its content but keep the site and its URL up and running. You'll also lose any modifications you've made to your current theme. This cannot be undone."
+					),
+					siteDomain
+				) }
+			</Text>
+			<ContentInfo siteContent={ siteContent } siteDomain={ siteDomain } />
+			<TextControl
+				__nextHasNoMarginBottom
+				__next40pxDefaultSize
+				label={ __( 'Type the site domain to confirm' ) }
+				value={ textValue }
+				onChange={ ( value ) => setTextValue( value.trim() ) }
+				help={ sprintf(
+					/* translators: %s: site domain */
+					__( 'The site domain is: %s' ),
+					siteDomain
+				) }
+			/>
+			<HStack spacing={ 4 } justify="flex-end">
+				<Button variant="tertiary" onClick={ onClose }>
+					{ __( 'Cancel' ) }
+				</Button>
+				<Button
+					variant="primary"
+					disabled={ textValue !== siteDomain }
+					isBusy={ isMutatePending }
+					onClick={ handleReset }
+				>
+					{ __( 'Reset site' ) }
+				</Button>
+			</HStack>
+		</VStack>
+	);
+}
+
+export default function SiteResetModal( { site, onClose }: { site: Site; onClose: () => void } ) {
+	const { createSuccessNotice } = useDispatch( noticesStore );
+	const [ error, setError ] = useState< string | null >( null );
+
+	const {
+		data: resetStatus,
+		isLoading: isLoadingResetStatus,
+		refetch: refetchResetStatus,
+	} = useQuery< SiteResetStatus >( siteResetStatusQuery( site.ID ) );
+	const { data: siteContent, isLoading: isLoadingSiteContent } = useQuery< SiteContent >(
+		siteResetContentSummaryQuery( site.ID )
+	);
+	const queryClient = useQueryClient();
+
+	const clearQueries = useCallback( () => {
+		queryClient.removeQueries( {
+			queryKey: siteResetContentSummaryQuery( site.ID ).queryKey,
+		} );
+
+		queryClient.removeQueries( {
+			queryKey: siteResetStatusQuery( site.ID ).queryKey,
+		} );
+	}, [ queryClient, site.ID ] );
+
+	const showSuccessNotice = useCallback( () => {
+		createSuccessNotice(
+			sprintf(
+				/* translators: %s: site domain */
+				__( '%s has been reset.' ),
+				site.slug
+			),
+			{ type: 'snackbar' }
+		);
+	}, [ createSuccessNotice, site.slug ] );
+
+	useEffect( () => {
+		if ( resetStatus?.status === 'completed' ) {
+			clearQueries();
+			showSuccessNotice();
+			onClose();
+		}
+	}, [ clearQueries, resetStatus?.status, showSuccessNotice, onClose ] );
+
+	if ( isLoadingSiteContent || isLoadingResetStatus ) {
+		return null;
+	}
```

Comment:
This check is redundant with the following `if ( ! siteContent || ! resetStatus )` check. If we use these `isLoading` from `useQuery()` just to return `null`, we can just use the next check. Basically, we don't need these `isLoading` variables.

Reply:
Yep. I left it in to discuss whether we need a loading state.